File: app/src/main/java/com/github/therajanmaurya/sweeterror/demo/SweetErrorInActivity.java
package com.github.therajanmaurya.sweeterror.demo;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.Snackbar;
import android.view.View;
import android.widget.LinearLayout;

import com.github.therajanmaurya.sweeterror.SweetErrorType;
import com.github.therajanmaurya.sweeterror.demo.base.SweetBaseActivity;
import com.github.therajanmaurya.sweeterror.demo.utils.ConstantKeys;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;


/**
 * @author Rajan Maurya
 *         On 05/11/17.
 */
public class SweetErrorInActivity extends SweetBaseActivity {

    @BindView(R.id.ll_sweet)
    LinearLayout llSweet;

    @BindView(R.id.layout_error)
    View layoutError;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sweet_error);
        ButterKnife.bind(this);
        initializeSweetUIErrorHandler(this, findViewById(android.R.id.content));

        SweetErrorType sweetErrorType = (SweetErrorType) getIntent()
                .getSerializableExtra(ConstantKeys.SWEET_ERROR_TYPE);

        llSweet.setVisibility(View.GONE);
        layoutError.setVisibility(View.VISIBLE);

        switch (sweetErrorType) {
            case EMPTY_UI:
                showSweetEmptyUI(getString(R.string.sweets), getString(R.string.sweet),
                        R.drawable.ic_adb_black_24dp);
                break;
            case ERROR_UI:
                showSweetErrorUI(getString(R.string.sweets));
                break;
            case NO_INTERNET:
                showSweetNoInternetUI();
                break;
            case CUSTOM:
                findViewById(R.id.fab_add_sweet).setVisibility(View.GONE);
                showCustomErrorUI(getString(R.string.no_sweets_found),
                        getString(R.string.come_later_again), R.drawable.ic_adb_black_24dp);
                break;
        }
    }

    @OnClick(R.id.fab_add_sweet)
    void addSweet() {
        Snackbar.make(findViewById(R.id.cl_sweet), "Adding Sweet :)",
                Snackbar.LENGTH_LONG).show();
    }

    @OnClick(R.id.btn_try_again)
    void retrySweet() {
        Snackbar.make(findViewById(R.id.cl_sweet), "Loading Sweet ...",
                Snackbar.LENGTH_LONG).show();
    }
}

File: app/src/main/java/com/github/therajanmaurya/sweeterror/demo/SweetErrorInActivity.kt
package com.github.therajanmaurya.sweeterror.demo

import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.res.ResourcesCompat
import com.github.therajanmaurya.sweeterror.SweetUIErrorHandler
import com.google.android.material.snackbar.Snackbar
import kotlinx.android.synthetic.main.activity_sweet_error.*


/**
 * @author Rajan Maurya
 * On 05/11/17.
 */
class SweetErrorInActivity : AppCompatActivity() {

    private lateinit var sweetUIErrorHandler: SweetUIErrorHandler

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sweet_error)

        sweetUIErrorHandler = SweetUIErrorHandler(this, findViewById(android.R.id.content))

        val sweetErrorType = intent
                .getSerializableExtra(ConstantKeys.SWEET_ERROR_TYPE) as SweetErrorType

        when (sweetErrorType) {
            SweetErrorType.EMPTY_UI -> {
                sweetUIErrorHandler.showSweetEmptyUI(getString(R.string.sweets),
                        getString(R.string.sweet), R.drawable.ic_adb_black_24dp,
                        llSweet, layoutError)
            }
            SweetErrorType.ERROR_UI -> {
                sweetUIErrorHandler.showSweetErrorUI(getString(R.string.sweets), llSweet, layoutError)
            }
            SweetErrorType.NO_INTERNET -> {
                sweetUIErrorHandler.showSweetNoInternetUI(llSweet, layoutError)
            }
            SweetErrorType.CUSTOM -> {
                fabAddSweet.hide()
                sweetUIErrorHandler.showSweetCustomErrorUI(getString(R.string.no_sweets_found),
                        getString(R.string.come_later_again), R.drawable.ic_adb_black_24dp,
                        llSweet, layoutError)
            }
            SweetErrorType.CUSTOM_STYLE -> {

                sweetUIErrorHandler.setBackgroundColor(R.color.grey_background)
                sweetUIErrorHandler.showCustomFeatureImage(false)
                sweetUIErrorHandler.setCustomFeatureImageTintColor(R.color.colorAccent)
                sweetUIErrorHandler.setCustomFeatureTextTypeFace(ResourcesCompat.getFont(this, R.font.nunito)!!)
                sweetUIErrorHandler.setCustomSubFeatureTextTypeFace(ResourcesCompat.getFont(this, R.font.nunito)!!)

                sweetUIErrorHandler.showSweetCustomErrorUI(getString(R.string.no_sweets_found),
                        getString(R.string.come_later_again), R.drawable.ic_adb_black_24dp,
                        llSweet, layoutError)
            }
        }


        fabAddSweet.setOnClickListener {
            Snackbar.make(clSweet, "Adding Sweet :)", Snackbar.LENGTH_LONG).show()
        }

        findViewById<Button>(R.id.btnTryAgain).setOnClickListener {
            Snackbar.make(clSweet, "Loading Sweet ...",
                    Snackbar.LENGTH_LONG).show()
        }
    }
}
